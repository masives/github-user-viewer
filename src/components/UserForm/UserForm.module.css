.form {
  border: 1px solid #ddd;
  border-radius: 0.25rem;
  display: block;
  margin: 0.75rem auto 0;
  max-width: 500px;
  padding: 1rem;
}

.label,
.input {
  display: block;
}

.input {
  border: 1px solid #ddd;
  border-radius: 0.25rem;
  font-size: 1rem;
  margin-bottom: 0.75rem;
  padding: 0.25rem;
  width: 100%;
}

.button {
  background: darkblue;
  border: 1px solid darkblue;
  border-radius: 0.25rem;
  color: white;
  cursor: pointer;
  display: block;
  font-size: 1.25rem;
  padding: 0.5rem;
}

.error {
  border-radius: 0.25rem;
  display: block;
  margin: 0.75rem auto 0;
  max-width: 500px;
  text-align: center;
  background: rgba(255, 0, 0, 0.1);
  border: 1px solid red;
  padding: 1rem;
}
