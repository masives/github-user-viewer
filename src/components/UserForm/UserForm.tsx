import * as React from 'react';

type PropTypes = {
  username: string;
  handleUserNameChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  handleFormSubmit: () => void;
};

const UserForm = ({ username, handleUserNameChange, handleFormSubmit }: PropTypes) => {
  const onFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    handleFormSubmit();
  };

  return (
    <form onSubmit={onFormSubmit}>
      <label>Username</label>
      <input name="username" type="text" value={username} placeholder="eg. masives" onChange={handleUserNameChange} />
    </form>
  );
};

export default UserForm;
