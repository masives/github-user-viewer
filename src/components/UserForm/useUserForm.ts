import { useState } from 'react';

const useUserForm = () => {
  const [username, setUsername] = useState<string>('');

  const handleUserNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setUsername(event.target.value);
  };

  return { username, handleUserNameChange };
};

export default useUserForm;
