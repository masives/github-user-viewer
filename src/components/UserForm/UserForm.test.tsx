import React from 'react';
import renderer from 'react-test-renderer';
import UserForm from './UserForm';

describe('UserForm component', () => {
  it('Should render correctly without error', () => {
    const component = renderer.create(
      <UserForm handleFormSubmit={() => {}} handleUserNameChange={() => {}} username="" errorMessage={null} />,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render correctly with error', () => {
    const component = renderer.create(
      <UserForm handleFormSubmit={() => {}} handleUserNameChange={() => {}} username="" errorMessage="Some error message" />,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
