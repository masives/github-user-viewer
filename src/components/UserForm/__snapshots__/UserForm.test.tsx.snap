// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm component Should render correctly with error 1`] = `
Array [
  <form
    className="form"
    onSubmit={[Function]}
  >
    <label>
      Username
      <input
        className="input"
        name="username"
        onChange={[Function]}
        placeholder="eg. masives"
        type="text"
        value=""
      />
      <button
        className="button"
        type="submit"
      >
        Submit
      </button>
    </label>
  </form>,
  <p
    className="error"
  >
    Some error message
  </p>,
]
`;

exports[`UserForm component Should render correctly without error 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <label>
    Username
    <input
      className="input"
      name="username"
      onChange={[Function]}
      placeholder="eg. masives"
      type="text"
      value=""
    />
    <button
      className="button"
      type="submit"
    >
      Submit
    </button>
  </label>
</form>
`;
